doc/user/content/sql/create-source/kafka.md

```diff
@@ -273,8 +273,8 @@ The following metadata is available for each source as a progress subsource:
 
 Field          | Type                                     | Meaning
 ---------------|------------------------------------------|--------
-`partition`    | `numrange`                               | The upstream Kafka partition.
-`offset`       | [`uint8`](/sql/types/uint/#uint8-info)   | The greatest offset consumed from each upstream Kafka partition.
+`partition`    | `numrange`                               | The upstream Kafka partitions.
+`offset`       | [`uint8`](/sql/types/uint/#uint8-info)   | The next possible offset to consume from the identified partitions, e.g. `offset - 1` shows the greatest offset we've consumed.
```

Comment:
I would either drop the `e.g` part because it becomes nonsensical for the special value zero which means that we haven't consumed anything